scripts/zstory.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+import psycopg
+import json
+from datetime import datetime
+from decouple import config
+
+"""
+This tool helps get an overview or the "story" of a certain zoom call.
+
+Something like
+<timestamp> slash-z meeting schedule link created
+<timestamp> zoom created, using license <blah>
+<timestamp> participant 1 joined
+<timestamp> participant 2 joined
+<timestamp> participant 1 left
+<timestamp> participant 2 left
+<timestamp> zoom destroyed, license <blah> released
+
+Requirements:
+- meeting name (e.g yzu82)
+- schedulingLink id 
+- get the license used by filtering for meetings using the schedulingLinkId in the Meeting table and look under ZoomID 
+- also copy the meetingId
+* Might also be useful to get the time that event happened
+- get the events by looking into the WebhookEvent table using the meetingId copied 
+- a zoom license is released when a zoom meeting is destroyed
+
+"""
+
+parser = argparse.ArgumentParser(
+    prog="zstory",
+    description="Helps you debug slash-z meetings"
+) 
+
+# the default argument we want to get is the scheduling link name
+parser.add_argument("meetid")
+
+# parse the arguments!
+args = parser.parse_args()
+
+print(f"Tracing meeting {args.meetid}...")
+
+# connect to the database
+with psycopg.connect(config("DATABASE_URL")) as conn:
+    # open cursor to perform database operations
+    with conn.cursor() as cursor:
+
+        # get the scheduling link id
+        cursor.execute('SELECT id FROM "SchedulingLink" WHERE name=%s', (args.meetid,)) # type: ignore
+        schedule = cursor.fetchone()
+        scheduling_link_id = schedule[0] if schedule else None
+
+        if scheduling_link_id is None:
+            print(f"Scheduling link with name {args.meetid} not found") 
+            quit()
+        
+
+        # query meeting id and zoom license
+        cursor.execute('SELECT (id, "zoomID") FROM "Meeting" WHERE "schedulingLinkId"=%s', (scheduling_link_id,))
+        meetings = cursor.fetchall()
+
+        print(f"\n{len(meetings)} meetings found")
+        for idx, meeting in enumerate(meetings):
```

Comment:
Just for my understanding - when searching for a meeting, it may be that we can find multiple?  Would this always be indicative of an error?

Reply:
afaik it should always return a list of records and I think it [should return an empty list](https://www.psycopg.org/psycopg3/docs/api/cursors.html#psycopg.Cursor.fetchall:~:text=defined%20by%20row_factory-,fetchall()%20%E2%86%92%20List%5BRow%5D,-%23) if there are no records

Reply:
not sure if multiple meetings with same id is expected wdy think @maxwofford ?